crates/uv-settings/src/settings.rs

```diff
@@ -484,29 +805,287 @@ pub struct PipOptions {
+        default = "[]",
+        value_type = "list[str]",
+        example = r#"
+            no-emit-package = ["ruff"]
+        "#
+    )]
     pub no_emit_package: Option<Vec<PackageName>>,
+    /// Include `--index-url` and `--extra-index-url` entries in the output file generated by `uv pip compile`.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-index-url = true
+        "#
+    )]
     pub emit_index_url: Option<bool>,
+    /// Include `--find-links` entries in the output file generated by `uv pip compile`.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-find-links = true
+        "#
+    )]
     pub emit_find_links: Option<bool>,
+    /// Include `--no-binary` and `--only-binary` entries in the output file generated by `uv pip compile`.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-binary = true
+        "#
+    )]
     pub emit_build_options: Option<bool>,
+    /// Whether to emit a marker string indicating the conditions under which the set of pinned
+    /// dependencies is valid.
+    ///
+    /// The pinned dependencies may be valid even when the marker expression is
+    /// false, but when the expression is true, the requirements are known to
+    /// be correct.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-marker-expression = true
+        "#
+    )]
     pub emit_marker_expression: Option<bool>,
+    /// Include comment annotations indicating the index used to resolve each package (e.g.,
+    /// `# from https://pypi.org/simple`).
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            emit-index-annotation = true
+        "#
+    )]
     pub emit_index_annotation: Option<bool>,
+    /// The style of the annotation comments included in the output file, used to indicate the
+    /// source of each package.
+    #[option(
+        default = "\"split\"",
+        value_type = "str",
+        example = r#"
+            annotation-style = "line"
+        "#
+    )]
     pub annotation_style: Option<AnnotationStyle>,
+    /// The method to use when installing packages from the global cache.
+    ///
+    /// Defaults to `clone` (also known as Copy-on-Write) on macOS, and `hardlink` on Linux and
+    /// Windows.
+    #[option(
+        default = "\"clone\" (macOS) or \"hardlink\" (Linux, Windows)",
+        value_type = "str",
+        example = r#"
+            link-mode = "copy"
+        "#
+    )]
     pub link_mode: Option<LinkMode>,
+    /// Compile Python files to bytecode after installation.
+    ///
+    /// By default, does not compile Python (`.py`) files to bytecode (`__pycache__/*.pyc`), instead
+    /// Python lazily does the compilation the first time a module is imported. In cases where the
```

Comment:
Perhaps "compilation will be performed lazily the first time a module is imported"